kcp/ckcp/setup-ckcp-on-openshift.sh

```diff
@@ -15,8 +15,54 @@ ARGOCD_MANIFEST="$SCRIPT_DIR/../../manifests/kcp/argocd/install-argocd.yaml"
 ARGOCD_NAMESPACE="gitops-service-argocd"
 
 cleanup() {
-  killall goreman
-  killall kubectl
+  echo "Running goreman run stop-all"
+  goreman run stop-all
+  echo "Running goreman run status"
+  goreman run status
+
+  proc=( goreman kubectl )
+  for SERVICE in "${proc[@]}"
+  do
+    echo "Sending SIGTERM and wait for 10 sec"
+    killall "$SERVICE"
+    sleep 10
+    if pgrep -x "$SERVICE" >/dev/null
+    then
+      echo "$SERVICE is still running."
+      echo "Sending SIGKILL"
+      killall -s SIGKILL "$SERVICE"
+    else
+      echo "$SERVICE stopped"
+    fi
+  done
+
+  echo "Check again for running leftovers:"
+  for SERVICE in "${proc[@]}"
+  do
+    if pgrep -x "$SERVICE" >/dev/null
+    then
+      echo "$SERVICE is zombie?"
+      echo "--- DEBUG ---"
+      ps -ef | grep "$SERVICE"
+      echo "--- DEBUG ---"
+
+      # Last effort
+      echo "Killing go processes"
+      pkill -9 go # should kill 'go run ./main.go' procs
+    fi
+  done
+
+  echo "Check for last time"
+  for SERVICE in "${proc[@]}"
+  do
+    if pgrep -x "$SERVICE" >/dev/null
+    then
+      echo "$SERVICE is still running..."
+      echo "Checking for Go processes..."
+      pgrep go
+    fi
+  done
+
```

Comment:
Thanks @drpaneas :rocket:  This looks good, just a suggestion instead of this can we just introduce a while() loop or wait.poll() which takes var serviceActive as a boolean, and unless it becomes false we keep running the inner loop until either it becomes false or timeouts and flag is still true;

Reply:
BTW fwiw lol I forgot this is in bash, so yeah same logic with bash options^

Reply:
This is a good suggestion in general but I'm not if favor of polishing this particular piece of code yet, simply because we do not know what is the root problem here (this code is trying to unravel tha problem). Let's find out first what causes the failure and then we can refactor this to be optimized for sure. Although there is no much gain here, because implementing an "until" loop in bash with booleans (bash doesn't like bools) it's going to complicate the code. This is running on the CI and worst case it will "waste" 10 seconds or 20 -- it's cheap and imho doesn't worth doing it. At least not now, until we know what works and what not.

Reply:
What we can do to not over-complicate things:

1. Create a separate cleanup.sh script.
2. From this script, call: `timeout 5m bash cleanup.sh`

WDYT?